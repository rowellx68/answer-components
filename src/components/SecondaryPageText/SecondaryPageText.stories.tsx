import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import SecondaryPageText, { SecondaryPageTextProps } from './SecondaryPageText'

export default {
  title: 'Components/Text/SecondaryPageText',
  component: SecondaryPageText,
  argTypes: {
    className: { control: 'text' },
    text: { control: 'text' },
  },
} as Meta;

const Template: Story<SecondaryPageTextProps> = (args) => (
  <SecondaryPageText {...args} />
)

export const BasicUsage = Template.bind({})
BasicUsage.args = {
  className: 'pb-6',
  text: 'This is a really long text that could be used as the sub-heading of the page.'
}

export const CustomFontSize = Template.bind({})
CustomFontSize.args = {
  className: 'pb-6 text-4xl md:text-6xl',
  text: 'This is a really long text that could be used as the sub-heading of the page.'
}
