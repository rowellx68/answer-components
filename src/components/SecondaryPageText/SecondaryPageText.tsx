import React from 'react'

export interface SecondaryPageTextProps {
  className?: string
  text: string
}

const SecondaryPageText: React.FC<SecondaryPageTextProps> = ({ text, className = '' }) => {
  return (
    <h2 className={`text-2xl md:text-3xl font-extralight ${className}`}>{text}</h2>
  )
}

export default SecondaryPageText
