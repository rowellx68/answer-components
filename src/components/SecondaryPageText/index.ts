import SecondaryPageText from './SecondaryPageText'
export default SecondaryPageText
