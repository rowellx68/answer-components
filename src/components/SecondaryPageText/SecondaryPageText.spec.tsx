import React from 'react'
import renderer from 'react-test-renderer'
import SecondaryPageText from './SecondaryPageText'

test('renders SecondaryPageText correctly', () => {
  const text = 'This is a really long text that could be used as the sub-heading of the page'
  const element = renderer
    .create(<SecondaryPageText text={text} />)
    .toJSON()

  expect(element).toMatchSnapshot()
})

test('renders SecondaryPageText correctly when passing className', () => {
  const text = 'This is a really long text that could be used as the sub-heading of the page'
  const element = renderer
    .create(<SecondaryPageText className="pb-6" text={text} />)
    .toJSON()

  expect(element).toMatchSnapshot()
})
