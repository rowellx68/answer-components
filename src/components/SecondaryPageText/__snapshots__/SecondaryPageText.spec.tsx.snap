// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SecondaryPageText correctly 1`] = `
<h2
  className="text-2xl md:text-3xl font-extralight "
>
  This is a really long text that could be used as the sub-heading of the page
</h2>
`;

exports[`renders SecondaryPageText correctly when passing className 1`] = `
<h2
  className="text-2xl md:text-3xl font-extralight pb-6"
>
  This is a really long text that could be used as the sub-heading of the page
</h2>
`;
